src/orchestrator/streaming.rs

```diff
@@ -43,133 +44,201 @@ use crate::{
-        debug!(?input_detections);
-        if input_detections.is_some() {
-            // Detected HAP/PII
-            // Do tokenization to get input_token_count
-            let (input_token_count, _tokens) =
-                tokenize(&ctx, model_id.clone(), input_text.clone()).await?;
-            // Send result with input detections
-            let _ = response_tx
-                .send(ClassifiedGeneratedTextStreamResult {
-                    input_token_count,
-                    token_classification_results: TextGenTokenClassificationResults {
-                        input: input_detections,
-                        output: None,
-                    },
-                    warnings: Some(vec![InputWarning {
-                        id: Some(InputWarningReason::UnsuitableInput),
-                        message: Some(UNSUITABLE_INPUT_MESSAGE.to_string()),
-                    }]),
-                    ..Default::default()
-                })
-                .await;
-        } else {
-            // No HAP/PII detected
-            // Do text generation (streaming)
-            let mut generation_stream =
-                generate_stream(&ctx, model_id.clone(), input_text.clone(), params.clone()).await?;
+        // Create response channel
+        #[allow(clippy::type_complexity)]
+        let (response_tx, response_rx): (
+            mpsc::Sender<Result<ClassifiedGeneratedTextStreamResult, Error>>,
+            mpsc::Receiver<Result<ClassifiedGeneratedTextStreamResult, Error>>,
+        ) = mpsc::channel(1024);
 
-            // Do output detections (streaming)
-            let output_detectors = task.guardrails_config.output_detectors();
-            match output_detectors {
+        tokio::spawn(async move {
+            // Do input detections (unary)
+            let masks = task.guardrails_config.input_masks();
+            let input_detectors = task.guardrails_config.input_detectors();
+            let input_detections = match input_detectors {
                 Some(detectors) if !detectors.is_empty() => {
-                    let aggregator = MaxProcessedIndexAggregator::default();
-                    let mut result_rx = streaming_output_detection_task(
-                        &ctx,
-                        detectors,
-                        aggregator,
-                        generation_stream,
-                    )
-                    .await?;
-                    // Forward generation results with detections to response channel
-                    tokio::spawn(async move {
-                        while let Some(generation_with_detections) = result_rx.recv().await {
-                            let _ = response_tx.send(generation_with_detections).await;
+                    match input_detection_task(&ctx, detectors, input_text.clone(), masks).await {
+                        Ok(result) => result,
+                        Err(error) => {
+                            error!(%request_id, %error, "task failed");
```

Comment:
We do log the actual error via it's `Display` impl (`%error`), but in this case we have a single `DetectorRequestError` variant that doesn't distinguish between input/output detection. Since this is existing behavior for the unary handler, I will the task of potentially expanding the error variants and/or improving logs to another PR.